src/neo4j_graphrag/experimental/components/entity_relation_extractor.py

```diff
@@ -100,28 +102,19 @@ def balance_curly_braces(json_string: str) -> str:
     return "".join(fixed_json)
 
 
-def fix_invalid_json(invalid_json_string: str) -> str:
-    # Fix missing quotes around field names
-    invalid_json_string = re.sub(
-        r"([{,]\s*)(\w+)(\s*:)", r'\1"\2"\3', invalid_json_string
-    )
-
-    # Fix missing quotes around string values, correctly ignoring null, true, false, and numeric values
-    invalid_json_string = re.sub(
-        r"(?<=:\s)(?!(null|true|false|\d+\.?\d*))([a-zA-Z_][a-zA-Z0-9_]*)\s*(?=[,}])",
-        r'"\2"',
-        invalid_json_string,
-    )
-
-    # Correct the specific issue: remove trailing commas within arrays or objects before closing braces or brackets
-    invalid_json_string = re.sub(r",\s*(?=[}\]])", "", invalid_json_string)
+def fix_invalid_json(raw_json: str) -> str:
+    repaired_json = json_repair.repair_json(raw_json)
 
-    # Normalize excessive curly braces
-    invalid_json_string = re.sub(r"{{+", "{", invalid_json_string)
-    invalid_json_string = re.sub(r"}}+", "}", invalid_json_string)
+    if isinstance(repaired_json, str):
+        repaired_json = repaired_json.strip()
+    else:
+        repaired_json = ""
 
-    # Balance curly braces
-    return balance_curly_braces(invalid_json_string)
+    if repaired_json.strip() == '""':
```

Comment:
It's already `strip`ed, right?

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/226/commits/85f2b450bd026074028f383670c30bd0cc5404fa